apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/Tasks/TestScheduler.cs

```diff
@@ -334,10 +334,47 @@ public override void Wait(Task task)
 
             public override async Task WhenAll(IEnumerable<Task> tasks)
             {
+                var exceptions = new List<Exception>();
+                var cancelled = false;
                 foreach (var task in tasks)
                 {
                     await this.ConfigureAwaitHideErrors(task);
+                    switch (task.Status)
+                    {
+                        case TaskStatus.RanToCompletion:
+                            break;
+                        case TaskStatus.Faulted:
+                            exceptions.AddRange(task.Exception.InnerExceptions);
+                            break;
+                        case TaskStatus.Canceled:
+                            cancelled = true;
```

Comment:
This needs to match the specification of [Task.WhenAll()](https://msdn.microsoft.com/en-us/library/hh160384(v=vs.110).aspx), so the returned Task must not complete until all tasks in the argument have completed, whatever the status of their completion (ie even if some cancel/fault, this method must still wait for all tasks to complete).